tests/nightly/gpu/test_t5.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Shamelessly copying relevant integration tests from HuggingFace.
+
+https://github.com/huggingface/transformers/blob/master/tests/test_modeling_t5.py
+
+Additionally includes some toy fine-tuning in ParlAI (similar to BART)
+"""
+
+import os
+import torch
+import unittest
+
+from parlai.agents.t5.task_specific_configs import TASK_CONFIGS
+from parlai.agents.t5.modules import set_device
+from parlai.core.agents import create_agent
+from parlai.core.params import ParlaiParser
+from parlai.core.torch_agent import Batch
+import parlai.utils.testing as testing_utils
+from parlai.utils.torch import padded_tensor
+from parlai.utils.testing import tempdir
+
+
+device = 'cpu' if not torch.cuda.is_available() else 'cuda:0'
+
+
+@testing_utils.skipUnlessGPU
+class TestT5Model(unittest.TestCase):
+    """
+    Test of T5 model.
+    """
+
+    def setUp(self):
+        opt = ParlaiParser(True, True).parse_args(['--model', 't5'])
+        self.agent = create_agent(opt)
+
+    @set_device
+    def test_small(self):
+        """
+        From Huggingface.
+        """
+        opt = ParlaiParser(True, True).parse_args(
+            ['--model', 't5', '--t5-model-arch', 't5-small', '--no-cuda']
+        )
+        agent_small = create_agent(opt)
+        text_vec = torch.LongTensor(agent_small.dict.txt2vec("Hello there")).unsqueeze(
+            0
+        )
+        label_vec = torch.LongTensor(agent_small.dict.txt2vec("Hi I am")).unsqueeze(0)
+
+        score = -agent_small.compute_loss(
+            Batch(text_vec=text_vec, label_vec=label_vec)
+        ) * label_vec.size(1)
+        EXPECTED_SCORE = -19.0845
+        self.assertAlmostEqual(score.item(), EXPECTED_SCORE, places=3)
+
+    @set_device
```

Comment:
uh what is this? lol

Reply:
has to do with how HF implements model parallel

Reply:
i can actually get rid of most of these in the testing file